### PR TITLE
Move dom element creation to js

### PR DESCRIPTION
**Pull request by gonzen2310**
#26

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@codecademy/40phaser",
+  "name": "40phaser",
```
Comment (JoshuaKGoldberg) [inline package-lock.json]: Agh I keep forgetting to do this... will push now 😄

File: src/scenes/end.js [hunk 2]
```diff
@@ -1,23 +1,82 @@
...
+        // add score text
+        this.add.text(options.windowWidth / 2 - 110, 60, `Score: ${score}`, {
+            fontFamily: "sans-serif",
+            fontSize: "48px",
+            fontStyle: "bold",
+            color: "#000",
+            align: "center",
+        });
+
+        // add subheding
```
Comment (JoshuaKGoldberg) [inline src/scenes/end.js]:
```suggestion
        // add subheading
```

...although I don't think these comments are super useful. A bunch of them have roughly the same information as member names, and the rest have text explaining what they are. Probably better to remove them.

File: src/scenes/end.js [hunk 3]
```diff
@@ -1,23 +1,82 @@
-import elements from "../elements.js";
+import options from "../options.js";
 
 export default class EndScene extends Phaser.Scene {
     constructor() {
         super({ key: "EndScene" });
+        this.gameReplay = {};
+        this.buildYourOwn = {};
```
Comment (JoshuaKGoldberg) [inline src/scenes/end.js]: I wouldn't recommend setting default values like this in the constructor unless those values will be used: which these are not. Here and in the other scenes.

IMO it's better to have no data then an inconsistent data shape.

File: src/scenes/end.js [hunk 4]
```diff
@@ -1,23 +1,82 @@
...
+            fill: "#000",
+        });
+
+        // add Play again button
+        this.gameReplay = this.add.rectangle(
+            options.windowWidth / 2,
+            options.windowHeight / 3 + 25,
+            210,
+            40,
+            0x6400e4,
```
Comment (JoshuaKGoldberg) [inline src/scenes/end.js]: These hex values are reused in several places. Could you move them into `options`? As with .scss files, it's convenient to have repeated/hardcoded values stored in one place.

Also, eventually we'll want them to be able to be overridden by whoever calls `launch40Phaser`.

File: src/scenes/start.js [hunk 5]
```diff
@@ -22,11 +24,88 @@ export default class StartScene extends Phaser.Scene {
         this.start_player.body.setAllowGravity(false);
         this.start_platform.body.setAllowGravity(false);
 
-        elements.buttonGameStart.onclick = () => {
+        this.createDOMElements();
+
+        this.gameStart.setInteractive();
+        this.buildYourOwn.setInteractive();
```
Comment (JoshuaKGoldberg) [inline src/scenes/start.js]: These `.setInteractive` calls feel like they should be part of `createDOMElements`, right? 

It's better to keep related responsibilities in one place. One shouldn't be able to call a method in another place and have it mysteriously not work as expected. "Principle of Least Astonishment" ❤️

File: src/options.js [hunk 6]
```diff
@@ -5,4 +5,13 @@ export default {
     platformSpeedIncrement: 0,
     windowHeight: 600,
     windowWidth: 800,
+    purpleBox: 0x6400e4,
```
Comment (JoshuaKGoldberg) [inline src/options.js]: Is there a reason some colors are in hex format (`0x6400e4`) and some are strings (`#6400e4`)? If not, let's standardize on strings. TIL you can use hex codes as colors!

File: src/options.js [hunk 7]
```diff
@@ -5,4 +5,13 @@ export default {
...
+    purpleText: "#6400e4",
+    whiteText: "#fff",
+    blackText: "#000",
+    backgroundColor: 0xf9f9f9,
+    smallFontSize: "17px",
+    mediumfFontSize: "24px",
```
Comment (JoshuaKGoldberg) [inline src/options.js]: Typo: `mediumfFontSize`

File: package-lock.json (hunk 1 above)
Comment (JoshuaKGoldberg) [reply package-lock.json]: Heh, you'll want to reset your `package-lock.json` to master now.

File: src/options.js (hunk 6 above)
Comment (LinKCoding) [reply src/options.js]: IIRC, Phaser has some funky/inconsistent means of adding color. But agreed, if there's a way to be consistent across our styling, I'm all for it :)

Comment (gonzen2310) [reply src/options.js]: Yeah, rectangle requires the color to be a number (0x000000) while text requires it to be a string("#00000")

Screen Shot 2019-07-03 at 10 19 55 AM

Comment (JoshuaKGoldberg) [reply src/options.js]: I was wrong. Cool 😄

**Pull request merged**